Add variants of FastZip.CreateZip that take filters as IScanFilter instead of strings

Numpsy opened the pull request:
refs #405: Add simple extra versions of FastZip.CreateZip that take file/directory filters as instances of IScanFilter rather than strings.

No unit tests yet (There don't seem to be any tests for ExtendedPathFilter either?)

A possible small api compatibility issue: if you have a call like

```csharp
zippy.CreateZip(tempZip.Filename, tempDir.Fullpath, true, null, null);
```
(as in an existing unit test I had to change), the two null literals for the filter parameters are ambiguous between the existing ```string``` overload and the new ```IScanFilter``` overload, which could possibly need a change to avoid?

_I certify that I own, and have sufficient rights to contribute, all source code and related material intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License._

Numpsy commented:
actually, I suppose there would also be a (sort of) conflict with #455 where that adds an extra overload with a leaveOpen param to the string-filter variants, and this would leave it without such an overload for the IScanFilter variants (hurrah for overload explision :-( )

piksel commented:
Yeah, but both variants are really nice to have. The `IScanFilter` overload could just take the leaveOpen with a default value instead, so it shouldn't be that bad.

Numpsy commented:
I'll get this updated tomorrow

Numpsy commented:
Rebased on top of the other changes, and changed the version of CreateZip that takes an output stream and the ScanFilters to have a ```leaveOpen``` parameter that defaults to false.
